lib/error/lib/merr.c

```diff
@@ -63,7 +63,7 @@ merr_file(merr_t err)
     if (err == 0 || err == -1)
         return NULL;
 
-    off = (int64_t)(err & MERR_FILE_MASK) >> MERR_FILE_SHIFT;
+    off = (int32_t)((int64_t)(err & MERR_FILE_MASK) >> MERR_FILE_SHIFT);
     if (off == 0)
```

Comment:
Make `off` a `uint64_t` and eliminate all casts.

Reply:
Breaks merr_test.

Reply:
I could make an int64_t if that is what you mean. That would reduce one cast.

Reply:
The offset field is a 16-bit field that should be treated as an integer.  The code could be written to be more clear about that, and maybe with comments.  As is, the casting just looks like random code to appease `-Wconversion`. 

Example that is more clear IMO:
```
    int16_t      off;

    /* Verify the "file" field is 16-bits
     */
    assert(UINT16_MAX == (UINT64_MAX & MERR_FILE_MASK) >> MERR_FILE_SHIFT);

    /* Extract into an int16_t var.
     */
    off = (int16_t)((err & MERR_FILE_MASK) >> MERR_FILE_SHIFT);
    if (off == 0)
        return NULL;

    file = hse_merr_base + (off * MERR_ALIGN);
```